### PR TITLE
Get `dev deploy` working in the CI service

### LINKED ISSUE #19: Enable `dev deploy` for the CPG instance
We'll need to deploy the `ci` service, but also remove any watched GitHub branches (we don't want to deploy whenever we push to `main`).

### PR DESCRIPTION
**Pull request by lgruen**
Fixes #19.

Removes a bunch of functionality that's only needed when watching GitHub branches, which we won't need for the CPG at this point.

### CODE REVIEW

File: ci/ci/ci.py [hunk 1]
```diff
@@ -27,8 +27,11 @@
 from .environment import BUCKET
 from .github import Repo, FQBranch, WatchedBranch, UnwatchedBranch, MergeFailureBatch
 
-with open(os.environ.get('HAIL_CI_OAUTH_TOKEN', 'oauth-token/oauth-token'), 'r') as f:
-    oauth_token = f.read().strip()
+oauth_token = None
+HAIL_CI_OAUTH_TOKEN = os.getenv('HAIL_CI_OAUTH_TOKEN')
+if HAIL_CI_OAUTH_TOKEN:
+    with open(HAIL_CI_OAUTH_TOKEN, 'r') as f:
+        oauth_token = f.read().strip()
```
Comment (illusional) [inline ci/ci/ci.py:35]: What's the behaviour on the startup when initialising the `gh_aiohttp.GitHubAPI` when the oauth_token is `None`:
https://github.com/populationgenomics/hail/blob/b6f287e891b39e61e8c76092855a15c9f1d5c864/ci/ci/ci.py#L311

**Review by illusional — COMMENTED** [review]
I haven't run it and I'm not as familiar with the usage, but there is code that still depends on the GH client (which requires the GH token). I'm not sure what the GitHub API will do if the token is not provided, will it just throw auth errors when the `dev_deploy_branch` is called?

https://github.com/populationgenomics/hail/blob/b6f287e891b39e61e8c76092855a15c9f1d5c864/ci/ci/ci.py#L248-L270

File: ci/ci/ci.py (hunk 1 above)
Comment (lgruen) [reply ci/ci/ci.py:21]: The OAuth token is optional. Cloning public branches (which is what we'd do for a `dev deploy`) works fine without a token.

I've cleaned up this file a bit more to only leave the functions that are necessary for `dev deploy`.

**Review by illusional — APPROVED** [review]
Looks good!

**Pull request merged**